lib/OptimizationBBO/src/OptimizationBBO.jl

```diff
@@ -181,7 +181,8 @@ function SciMLBase.__solve(cache::Optimization.OptimizationCache{
 
     t1 = time()
 
-    opt_ret = Symbol(opt_res.stop_reason)
+    # Use the improved convert function
+    opt_ret = isa(opt_res.stop_reason, String) ? Optimization.deduce_retcode(opt_res.stop_reason) : Optimization.deduce_retcode(Symbol(opt_res.stop_reason))
```

Comment:
What other type does the `stop_reason` take than String?

Reply:
I did this cause of the Cl error, but we don't really need to do type-checking.

Reply:
```suggestion
    opt_ret = Optimization.deduce_retcode(opt_res.stop_reason)
```